File: server/models/Category.js
const mongoose = require("mongoose");

const categorySchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    name: {
      type: String,
      required: [true, "Category name is required"],
      trim: true,
    },
    type: {
      type: String,
      required: true,
      enum: ["working-hours", "skills", "schedule", "timetable"],
      default: "working-hours",
    },
    color: {
      type: String,
      default: "#3498db",
    },
    icon: {
      type: String,
      default: "circle",
    },
    description: {
      type: String,
      trim: true,
    },
  },
  {
    timestamps: true,
  }
);

categorySchema.index({ user: 1, name: 1, type: 1 }, { unique: true });

module.exports = mongoose.model("Category", categorySchema);

File: backend-go/internal/models/category.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type CategoryType string

const (
	CategoryTypeWorkingHours CategoryType = "working-hours"
	CategoryTypeSkills       CategoryType = "skills"
	CategoryTypeSchedule     CategoryType = "schedule"
	CategoryTypeTimetable    CategoryType = "timetable"
)

type Category struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	User        primitive.ObjectID `bson:"user" json:"user"`
	Name        string             `bson:"name" json:"name"`
	Type        CategoryType       `bson:"type" json:"type"`
	Color       string             `bson:"color" json:"color"`
	Icon        string             `bson:"icon" json:"icon"`
	Description string             `bson:"description,omitempty" json:"description,omitempty"`
	CreatedAt   time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt   time.Time          `bson:"updatedAt" json:"updatedAt"`
}
